src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Legacy/Tokenizer.cs

```diff
@@ -119,7 +118,7 @@ SyntaxToken ITokenizer.NextToken()
         return default(SyntaxToken);
     }
 
-    public void Reset()
+    public virtual void Reset(int position)
```

Comment:
Should the `position` parameter be used inside this method?

Reply:
No, it's not needed in the base type. Only in the derived type.

Reply:
Should we assert that it's never passed here then? E.g., that it's equal to `StartState` or making it `int? position = null` and asserting that it's `null` here.

Reply:
I don't think we can do any of that. The reason that we need this for the roslyn-based tokenizer is that the roslyn-based tokenizer needs to be kept in sync as well. The other tokenizers don't have to do that work, and don't care what position is passed. They just go off what the SeekableTextReader says is the current position. I considered moving all of the Reset code into here, but that felt like more trouble that it was worth.